CMakePCHCompiler.cmake

```diff
@@ -111,22 +125,31 @@ function(target_precompiled_header) # target [...] header
 			endif()
 		endif()
 
-		add_dependencies(${target} ${pch_target})
-
 		if(MSVC)
-			get_filename_component(win_pch "${target_dir}/${header}.pch" ABSOLUTE)
-			get_filename_component(win_header "${header}" ABSOLUTE)
+			#Note, this adds debug/release/etc. onto the end.
+			set_target_properties(${target} PROPERTIES
+				COMPILE_PDB_OUTPUT_DIRECTORY ${pdb_dir}) ##/Fd\"${pdb_dir}\\\\\"
+			#Careful: set_target_properties is destructive
```

Comment:
I explained this to @gjasny ... basically don't use `set_target_properties` in a module, since it messes up the user's script.

Reply:
On second thought, I'm not sure what this means. My best guess is it refers to `set(exclude "/Yu${win_header}")` since that is the line under the comments. I think it is adding a variable in order to repair things later on.... where probably `set_target_properties` is used with `exclude`.